sql/table.go

```diff
@@ -365,8 +393,35 @@ func convertDatum(col ColumnDescriptor, val parser.Datum) (interface{}, error) {
 			return float64(v), nil
 		}
 	// case ColumnType_DECIMAL:
-	// case ColumnType_DATE:
-	// case ColumnType_TIMESTAMP:
+	case ColumnType_DATE:
+		switch val := val.(type) {
+		case parser.DString:
```

Comment:
Oh yes, this was some remnants of implicit typing that I was trying out. I've removed this code completely. Thanks!